### PR TITLE
Fix Android compilation and some tests

### PR DESCRIPTION
**Pull request by kumpera**
cc @luhenry

### CODE REVIEW

File: mcs/class/System/Test/System.Diagnostics/ProcessTest.cs [hunk 1]
```diff
@@ -1124,6 +1124,7 @@ public void GetProcessesByName()
 
 		[Test]
 		[ExpectedException(typeof(InvalidOperationException))]
```
Comment (luhenry) [inline mcs/class/System/Test/System.Diagnostics/ProcessTest.cs]: This test fails on CI

**Comment by kumpera** [comment]
I spend the day trying to fix that test on OSX but, by design, you can't get init's process name without special privilege. So I'm disabling it instead. :(

**Comment by akoeplinger** [comment]
The point of the test _was_ to verify we get an exception when accessing the init process name: ecf25101086810b83d9d4d30ee532c3e2b56ec9b

If it works on Linux/Android, maybe we should constrain the test to OSX?

**Comment by kumpera** [comment]
The test is too fragile, it's a bad idea to depend on such OS specific details that do change all the time.

**Pull request merged**